### ISSUE TITLE
Allow NaN in batch fitting

### ISSUE THREAD

**Issue by Jhsmit** [Jhsmit/PyHDX#174]
Make the torch.mean calls NaN compatible such that regularizers can be computed without introducing zero padding

**Comment by Jhsmit** [comment]
Torch recently added nanmean: https://pytorch.org/docs/stable/generated/torch.nanmean.html
